Parsers/ASimAuditEvent/Parsers/vimAuditEventIllumioSaaSCore.yaml

```diff
@@ -0,0 +1,451 @@
+              isnotnull(temp_action), temp_action.src_ip,
+              ''
+          )
+      | extend 
+          ActorUsername = case(
+              isnotnull(created_by.system), "System",
+              isnotnull(created_by.user), created_by.user.username,
+              isnotnull(created_by.agent), created_by.agent.hostname,
+              "Unknown"
+          )          
+      | extend 
+          ActorUsernameType = case(
+              isnotnull(created_by.system), "System",
+              isnotnull(created_by.user), "User",
+              isnotnull(created_by.agent), "Agent",
+              "Unknown"
+          )                      
+      // ***** parser filter params *****
+      | where ((isnull(starttime) or TimeGenerated >= starttime) and (isnull(endtime) or TimeGenerated <= endtime))             
+          and (array_length(eventtype_in) == 0 or EventType in (eventtype_in))          
+      | where (eventresult == "*" or EventResult =~ eventresult) and (array_length(actorusername_has_any) == 0 or ActorUsername has_any (actorusername_has_any)) 
+      and (array_length(srcipaddr_has_any_prefix) == 0 or  has_any_ipv4_prefix(SrcIpAddr, srcipaddr_has_any_prefix))
+      and (array_length(operation_has_any) == 0 or Operation has_any (operation_has_any))
+      and (array_length(object_has_any) == 0 or Object has_any (object_has_any))
+      //and (array_length(newvalue_has_any) == 0 or NewValue has_any (newvalue_has_any))
+      // ***** parser filter params *****
+      | extend
+          EventCount = int(1),
+          EventStartTime = TimeGenerated, 
+          EventEndTime= TimeGenerated,
+          EventProduct = 'Core',
+          EventVendor = 'Illumio',
+          EventSchemaVersion = '0.1.0',
+          EventSchema = 'AuditEvent',
+          Dvc = pce_fqdn,
+          EventType = iff(isnull(EventType), event_type, EventType),
+          EventOriginalUid = href
+      //aliases
+      | extend          
+          IpAddr = SrcIpAddr,
+          User = ActorUsername,
+          Value = NewValue
+      | project-away 
+          event_type, // used by EventType          
+          severity, // used by EventSeverity  
+          temp_*,      
+          resource_changes, // used by NewValue and EventMessage
+          notifications,
+          version, // simply drop version, no need to translate
+          action, //used by src_ip
+          status, // used by EventResult
+          created_by, // used by ActorUsername and ActorType
+          pce_fqdn, // used by Dvc
+          href, // used by EventOriginalUid
+          TenantId
+  };
+  parser (
+      starttime = starttime,
+      endtime = endtime,
+      srcipaddr_has_any_prefix = srcipaddr_has_any_prefix,
+      actorusername_has_any = actorusername_has_any,
+      eventtype_in = eventtype_in,
+      eventresult = eventresult,
+      operation_has_any = operation_has_any,
+      object_has_any=object_has_any,
+      newvalue_has_any=newvalue_has_any,
```

Comment:
would it be better to enumerate all the possible new value keys and based on a case statement, extract the new value?
I wanted to avoid the above approach and store the whole json object to preserve context of the entire object.
If you notice the first record, there are multiple keys, log_traffic, visibility_level, enforcement_mode and so on.
Each have a before/after key, what would be your approach to extract them?

Reply:
sounds good.